toolbox/fdc3-conformance/Metadata-Tests.md

```diff
@@ -0,0 +1,62 @@
+| Test   | 4.RaiseIntent   | Use `appIdentifier1` to raise an intent and target that instance, with<br/> `const resolution = fdc3.raiseIntent("aTestingIntent", {"type": "testContextX"}, appIdentifier1)` |
+| Test   | 5.Confirm1 | Check that `resolution.source` matches `appIdentifier1` |
+| A | 6.ConfirmReceipt | Ensure that the instance of app A represented by `appIdentifier1` received the raised intent |
+
+- `FindInstances`: Perform the above steps
+
+## Getting Info For The Agent
+
+| App | Step           | Details                                                                                           |
+|-----|----------------|---------------------------------------------------------------------------------------------------|
+| Test   | 1.getInfo    |Retrieve the `ImplementationMetadata` for the DesktopAgent with <br/> `fdc3.getInfo().then((implMetadata) => {  ... subsequent steps ...}`<br>**Note that the use of `then` is deliberate and intended to confirm that a promise returned (as this function switched from synchronous to asynchronous in 2.0)**|
+| Test   | 2.CheckVersion  | Check that the `fdc3Version` variable is present and at or greater than 2.0 (which you can do with the [`versionIsAtLeast` function from FDC3's Methods.ts](https://github.com/finos/FDC3/blob/add64f8302c6dcdc8437cf0e245101e927b69ec2/src/api/Methods.ts#L207):<br>`const isFDC3v2 = versionIsAtLeast(implMetadata, "2.0")`  |
+| Test   | 3.CheckProvider  | Check that the `provider` variable is present and not an empty string  |
+| Test   | 4.CheckFeatures  | Check that the `optionalFeatures`, `optionalFeatures.OriginatingAppMetadata` and `optionalFeatures.UserChannelMembershipAPIs` variables are all present and that the latter two provide boolean values  |
+
+- `GetInfo1`: Perform the above steps
+
+| App | Step           | Details                                                                                           |
+|-----|----------------|---------------------------------------------------------------------------------------------------|
+| Test   | 1.Open1    | Start an instance of App A with <br/>`const appIdentifier1 = await fdc3.open({appId: "<A's appId>"})` <br /> retrieve its `AppIdentifier` with instance details. Confirm that the `AppIdentifier` contains both an `appId` and `instanceId` |
+| A | 2.getInfo     |   Retrieve the `ImplementationMetadata` for the DesktopAgent with: <br  />`fdc3.getInfo().then((implMetadata) => {  ... subsequent steps ...}` <br/> This should include `AppMetadata` for the retrieving app. |
+| A + Test | 3.Confirm  | Check that `implMetadata.appMetadata` contains an `appId` and `instanceId` matching that retrieved in the first step (will require transmission of the details from A to Test or vice-versa). Also compare the `AppMetadata` object to the expected definition for the fields provided above during setup and ensure that the metadata matches. |
+
+- `GetInfo2`: Perform the above steps.
```

Comment:
Note this test only applies to 2.0. 

```suggestion
| App | Step           | Details                                                                                           |
|-----|----------------|---------------------------------------------------------------------------------------------------|
| Test   | 1.Open1    | Start an instance of App A with <br/>`const appIdentifier1 = await fdc3.open({appId: "<A's appId>"})` <br /> retrieve its `AppIdentifier` with instance details. Confirm that the `AppIdentifier` contains both an `appId` and `instanceId` |
| A | 2.getInfo     |   Retrieve the `ImplementationMetadata` for the DesktopAgent with: <br  />`fdc3.getInfo().then((implMetadata) => {  ... subsequent steps ...}` <br/> This should include an `appMetadata` field with the `AppMetadata` for the retrieving app. |
| A + Test | 3.Confirm  | Check that `implMetadata.appMetadata` contains an `appId` and `instanceId` matching those retrieved in the first step (will require transmission of the details from A to Test or vice-versa). Also compare the `AppMetadata` object to the expected definition for the fields provided above during setup and ensure that the metadata matches. |

- `GetInfo2`: ![2.0](https://img.shields.io/badge/FDC3-2.0-blue) Perform the above steps.
```

Reply:
At the top of the Metadata-Tests.md file there is a big "2.0" badge, indicating that the whole lot only applies to 2.0
However, if we make `GetInfo` apply to both versions I'll change this and add the badges to the individual tests in the file

Reply:
👍

Reply:
@Gaganahluwalla you'll need to make this change on your side too

Reply:
@robmoffat - Does it mean a new Advanced FDC3.getInfo Test case is needed in 1.2?

Reply:
yes, that's right